Adjudicator support

atinm opened the pull request:
Added an Adjudicator mode checkbox. When in Adjudicator mode, new buttons are available for Validate (keyboard shortcut V), Invalidate (keyboard shortcut I), Validate all Unvalidated, Invalidate all Unvalidated lines.

The adjudicator's changes are stored in a *.adjudication.json file rather than a per rater file and that file contains all annotations and all lines. When in adjudication mode, if a *.adjudication.json exists, that is the only file that is read, otherwise we read the rater files as usual but save to *.adjudication.json. The annotations schema has been updated to allow lines to have adjudicator, the status and a timestamp. Most users will not use the adjudication, but we will be able to see the adjudications, or hide invalidated lines if desired after agreement.

dqdinh commented:
I'm playing around with the Adjudicator mode and have some initial thoughts from a UX perspective. I'm not the end user so feel free to disregard any of the following:

- An option to reset or clear all adjudicated lines in a frame.

- An option to toggle all annotations on and off. Use case is the Adjudicator is trying to see the pleura line artifact but annotation likes are getting in the way.

- The `Save & next` button saves the Adjudicator json but does not go to the next dicom file.

- It's difficult to differentiate between valid and invalid lines after marking them. For example, if I've marked a line as valid but  later changed my mind, then I should be able to hover over the line to see a label above the line or status text at the bottom telling me if the line is valid or invalid.

- When I use the Next / Prev Visible, I'm able to change a valid line to invalid and vice versa but I'm not able to change a valid line to be a duplicate line.

- When I press Next Line or Prev Line, it would be helpful to have another visual marker in addition to line color to show current line selected. For example, a small cursor or arrow point at the line midpoint to showing me the current line selection.

- In addition to hiding the rater selection when entering Adjudicator mode, would also make sense to hide the `Sector annotations` UI section since the Adjudicator is not expected to add or remove any annotations. We can add button in the Adjudication tools for the Overlay option since that might be used by the Adjudicator to when Adjudicating.

- Not sure what I did but the pleura overlay is on but no pleura lines. Currently all lines have been adjudicated.
![Screenshot 2025-06-25 at 11 21 02 PM](https://github.com/user-attachments/assets/44f04bc2-9e4d-4561-b636-4b582dfb978c)

- Counter show the current number of unadjudicated, valid, invalid, and duplicated lines.

atinm commented:
Added:
1. Reset all adjudications.
2. Show/Hide lines
3. Fixed Save and Next the broke in one of the changes because we were checking the return code of SaveAnnotations() in SaveAndLoadNext() but it wasn't returning anything explicitly
4. Changed the endpoint to Cross2D for invalidated and lowered opacity so they look more faded and distinct from duplicated lines which use Diamond2D and are a little brighter, and unadjudicated which are brighter still.
5. Can change status to between valid, invalid, duplicate freely (but not back to unadjudicated - except through Reset all)
6. Nick thought he would want to add his own lines, so leaving the Sector annotations - maybe can minimize them by default when in adjudication mode
7. Haven't been able to duplicate the missing all lines thing you saw.
8. I am not sure about the stats - the status text goes away, and really the only important number is the unadjudicated because once it is fully adjudicated, the only lines that matter are the valid ones.

dqdinh commented:
1. When I click `Save & next` or the next arrow, the rater selection section becomes uncollapsed and reveals the annotators for the DICOM file that is getting loaded. The sector annotations section also becomes uncollapsed so the this issue impacts all pre-collapsed UI sections.
![Screenshot 2025-06-26 at 2 47 53 PM](https://github.com/user-attachments/assets/56b5fc91-7d33-45b0-b46e-76f7ca0c9204)

2. Clicking on  Invalidate or Duplicate rest buttons doesn't change the node shapes (they stay the same circle node shape as validate). This is different behavior then using the Invalidate Line and Mark Duplicate buttons.
![Screenshot 2025-06-26 at 2 50 40 PM](https://github.com/user-attachments/assets/026324d5-71f6-4430-adfd-da1f6036245c)
![Screenshot 2025-06-26 at 2 50 48 PM](https://github.com/user-attachments/assets/0828cb7c-62f2-464c-a6a5-0337510ca8bd)



4. Clicking Save button does not show any status text providing UI feedback on whether the save is successful.

dqdinh reviewed (commented):
UX lgtm! 

Just one comment about dup test code.

atinm commented on AnnotateUltrasound/Testing/Python/AnnotateUltrasoundTest.py:
```diff
@@ -0,0 +1,68 @@
+from slicer.ScriptedLoadableModule import *
+from AnnotateUltrasound import AnnotateUltrasoundLogic
+
+class AnnotateUltrasoundTest(ScriptedLoadableModuleTest):
```
Done
